plugins/chat/assets/javascripts/discourse/components/chat-footer.gjs

```diff
@@ -0,0 +1,90 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import i18n from "discourse-common/helpers/i18n";
+import eq from "truth-helpers/helpers/eq";
+
+export default class ChatFooter extends Component {
+  @service router;
+  @service chat;
+  @service chatApi;
+
+  @tracked threadsEnabled = false;
+
+  constructor() {
+    super(...arguments);
+    this.userThreadCount();
+  }
+
+  async userThreadCount() {
+    try {
+      const result = await this.chatApi.userThreadCount();
+      this.threadsEnabled = result.thread_count > 0;
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
```

Comment:
I would try to implement this as a modifier:

```
updateThreadCount = modifier(() => {
  let result;
   try {
      result = await this.chatApi.userThreadCount();
      this.threadsEnabled = result.thread_count > 0;
    } catch (error) {
      popupAjaxError(error);
    }
    
    return () => {
      result?.abort() // this is the nice part, when the component is destroyed, it will abort the current request
    }
});
```

and then

```
<nav class="c-footer" {{this.updateThreadCount}}>
```

Reply:
If it's not working with a modifier (maybe because of async) then I would still try to hold a reference to the request to cancel it if we destroy the component.

Reply:
yes that should be called automatically when the DOM node is inserted